CMakeLists.txt

```diff
@@ -197,6 +197,40 @@ target_compile_definitions(celerity_runtime PUBLIC
   CELERITY_FEATURE_LOCAL_ACCESSOR=$<BOOL:${CELERITY_FEATURE_LOCAL_ACCESSOR}>
 )
 
+# Generate header containing version information.
+#
+# We want this command to run before each build. The canonical hack for doing
+# this is to add a second output file that is not actually being created.
+# The timestamp on the generated header file is only updated when its configured
+# contents change, so this will not cause unnecessary re-compilations.
+#
+# However, this does not work for the Unix Makefiles generator, as it will
+# for some reason always update the timestamp on version.h, causing a rebuild
+# of all translation units that depend on it.
+# As a fallback, we specify the .git folder as a dependency.
+set(VERSION_HEADER_OUTPUT_FILES ${CMAKE_CURRENT_LIST_DIR}/include/version.h)
+if(NOT CMAKE_GENERATOR STREQUAL "Unix Makefiles")
+  list(APPEND VERSION_HEADER_OUTPUT_FILES "_force_rerun")
+else()
+  # Fallback for Unix Makefiles: Re-run custom command when timestamp
+  # on .git folder changes (which happens after most (all?) git interactions).
+  if(IS_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}/.git)
+    set(VERSION_HEADER_MAKEFILE_WORKAROUND ${CMAKE_CURRENT_LIST_DIR}/.git)
+  endif()
+endif()
+add_custom_command(
```

Comment:
This workaround becomes unnecessary when we replace `add_custom_command` with `add_custom_target` directly: Custom targets are always out of date and do not have any output names, so `make` won't touch the file as long as `configure_file()` in the generator script doesn't.

Reply:
Yes, much cleaner!